tests/acceptance/shared/cluster.go

```diff
@@ -0,0 +1,317 @@
+package shared
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+var (
+	KubeConfigFile string
+	AwsUser        string
+	AccessKey      string
+)
+
+type Node struct {
+	Name       string
+	Status     string
+	Roles      string
+	Version    string
+	InternalIP string
+	ExternalIP string
+}
+
+type Pod struct {
+	NameSpace string
+	Name      string
+	Ready     string
+	Status    string
+	Restarts  string
+	NodeIP    string
+	Node      string
+}
+
+const (
+	GrepNoExec    = " | grep -v NoSchedule | grep -v NoExecute"
+	RunningAssert = "Running"
+)
+
+// ManageWorkload creates or deletes a workload based on the action: create or delete.
+func ManageWorkload(action, workload string) (string, error) {
+	if action != "create" && action != "delete" {
+		return "", fmt.Errorf("invalid action: %s. Must be 'create' or 'delete'", action)
+	}
+	var res string
+	var err error
+
+	resourceDir := BasePath() + "/acceptance/workloads/"
+
+	files, err := os.ReadDir(resourceDir)
+	if err != nil {
+		err = fmt.Errorf("%s : Unable to read resource manifest file for %s", err, workload)
+		return "", err
+	}
+
+	for _, f := range files {
+		filename := filepath.Join(resourceDir, f.Name())
+		if strings.TrimSpace(f.Name()) == workload {
+			if action == "create" {
+				res, err = createWorkload(workload, filename)
+				if err != nil {
+					return "", fmt.Errorf("failed to create workload %s: %s", workload, err)
+				}
+			} else {
+				err = deleteWorkload(workload, filename)
+				if err != nil {
+					return "", fmt.Errorf("failed to delete workload %s: %s", workload, err)
+				}
+			}
+			return res, err
+		}
+	}
+
+	return "", fmt.Errorf("workload %s not found", workload)
+}
+
+// createWorkload creates a workload.
+func createWorkload(workload, filename string) (string, error) {
+	fmt.Println("\nDeploying", workload)
+	return RunCommandHost("kubectl apply -f " + filename + " --kubeconfig=" + KubeConfigFile)
+
+}
+
+// deleteWorkload deletes a workload and asserts that the workload is deleted.
+func deleteWorkload(workload, filename string) error {
+	fmt.Println("\nRemoving", workload)
+	cmd := "kubectl delete -f " + filename + " --kubeconfig=" + KubeConfigFile
+
+	_, err := RunCommandHost(cmd)
+	if err != nil {
+		return fmt.Errorf("failed to run kubectl delete: %v", err)
+	}
+
+	timeout := time.After(60 * time.Second)
+	tick := time.Tick(5 * time.Second)
+
+	for {
+		select {
+		case <-timeout:
+			return errors.New("workload deletion timed out")
+		case <-tick:
+			isDeleted, err := IsWorkloadDeleted(workload)
+			if err != nil {
+				fmt.Printf("failed to check if workload is deleted: %v", err)
+			} else if isDeleted {
+				return nil
+			}
+		}
+	}
+}
+
+// IsWorkloadDeleted returns true if the workload is deleted.
+func IsWorkloadDeleted(workload string) (bool, error) {
```

Comment:
Merge this functionality into deleteWorkload

Reply:
done